components/profiler/Cargo.toml

```diff
@@ -0,0 +1,21 @@
+[package]
+name = "profiler"
+version = "0.0.1"
+edition = "2018"
+publish = false
+
+[features]
+profiling = ["lazy_static", "cpuprofiler", "callgrind", "valgrind_request"]
+
+[dependencies]
+tikv_alloc = { path = "../tikv_alloc" }
+
+[target.'cfg(unix)'.dependencies]
+lazy_static = { version = "1.2.0", optional = true }
+cpuprofiler = { version = "0.0.3", optional = true }
+callgrind = { version = "1.1.0", optional = true }
```

Comment:
Is valgrind a useful use case? I think most of the time only cpuprofiling is used.

Reply:
valgrind is very useful for micro benchmark, which can report precise amount of function calls, as well as precise (emulated) cache hit.

Reply:
I'm also hoping that this crate can be extended into a general purpose profiling module that can be published for the community, and ultimately fulfill the various requirements of the Go profiling tools. More profiler options in that case seems good.